iceoryx_hoofs/memory/include/iox/detail/scoped_static.inl

```diff
@@ -25,7 +25,7 @@ template <typename T, typename... CTorArgs>
 inline ScopeGuard makeScopedStatic(T& memory, CTorArgs&&... ctorArgs) noexcept
 {
     memory.emplace(std::forward<CTorArgs>(ctorArgs)...);
-    return ScopeGuard([&memory]() { memory.reset(); });
+    return ScopeGuard([&memory]() noexcept { memory.reset(); });
```

Comment:
Well, this is only `noexcept` if the underlying type stored does not throw. You could add a todo here, to derive the exception specifier from `T`.

Reply:
Something for the future :)
Will add the todo in the next PR